vars/provisionNodes.groovy

```diff
@@ -121,10 +121,17 @@ def call(Map config = [:]) {
                                              python-debuginfo python2-aexpect libcmocka          \
                                              python-pathlib python2-numpy git                    \
                                              golang-bin'''
-    sh script: 'set -x; rm -f ci_key*; ssh-keygen -N "" -f ci_key;' +
-               ' pdcp -R ssh -l root -w ' + nodeString +
-               ' ci_key* /tmp/;' +
-               ' pdsh -R ssh -l root -w ' + nodeString +
-               ' "' + provision_script + '" 2>&1 | dshbak -c'
+    def rc = 0
+    rc = sh(script: 'set -x; rm -f ci_key*; ssh-keygen -N "" -f ci_key;' +
+                    ' pdcp -R ssh -l root -w ' + nodeString +
+                    ' ci_key* /tmp/;' +
+                    ' pdsh -R ssh -l root -w ' + nodeString +
+                    ' "' + provision_script + '" 2>&1 | dshbak -c',
```

Comment:
This will return the return code of dshbak, not the provision_script which I doubt it what you expect or want.

Reply:
Hoist with his own petard.

Well, kinda sorta.  I wasn't quite bitten by `$?` vs `${PIPESTATUS[0]}` but this is a pitfall I point out frequently.